nex.go

```diff
@@ -595,7 +602,12 @@ func gen(out *bufio.Writer, x *rule) {
 		v := todo[len(todo)-1]
 		todo = todo[0 : len(todo)-1]
 		// Singles.
+		var runes []rune
 		for r, _ := range sing {
+			runes = append(runes, r)
+		}
+		sort.Sort(RuneSlice(runes))
+		for _, r := range runes {
```

Comment:
This is the fix right here.  The previous `range sing` is an iteration over map (random order).  By iterating over sorted runes, it's deterministic.